### ISSUE TITLE
Use Forecast Extension instead of package-specific fields

### ISSUE THREAD

**Issue by m-mohr** [stactools-packages/ecmwf-forecast#8]
The fields 
>    "ecmwf:reference_datetime": "2022-02-01T00:00:00Z",
>    "ecmwf:forecast_datetime": "2022-02-01T00:00:00Z",
>    "ecmwf:step": "0h",

are also defined in the forecast extension. The fields should be preferred over the fields used here.

**Comment by TomAugspurger** [comment]
Agreed. We'll probably want the `forecast:` fields in addition to these, to avoid breaking users relying on the current names.
